src/main/java/org/hawkular/client/android/backend/model/MetricAvailabilityBucket.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.client.android.backend.model;
+
+import com.google.gson.annotations.SerializedName;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+
+public class MetricAvailabilityBucket implements Parcelable {
```

Comment:
Actually no. I chose bad naming but 
Gauge and counter bucket send data in same fashion but Availability bucket is completely different. So class named MetricAvailabilityBucket only for Availabilility and MetricBucket for Counter and Gauge Buckets